app/src/main/java/com/hamdan/forzenbook/domain/usecase/login/LoginGetTokenFromDatabaseUseCaseImpl.kt

```diff
@@ -0,0 +1,15 @@
+package com.hamdan.forzenbook.domain.usecase.login
+
+import android.util.Log
+import com.hamdan.forzenbook.data.repository.LoginRepository
+
+class LoginGetTokenFromDatabaseUseCaseImpl(
+    val repository: LoginRepository
+) : LoginGetTokenFromDatabaseUseCase {
+    override suspend operator fun invoke() {
+        val token = repository.getToken()?.token ?: throw Exception("null token")
```

Comment:
Make your own exception for this like in the other spots

Reply:
created, makes sense to have the specific exception will make handling the error easier down the line when I actually implement sending to the homepage